yt_dlp/extractor/common.py

```diff
@@ -1028,16 +1031,19 @@ def __decode_webpage(self, webpage_bytes, encoding, headers):
         except LookupError:
             return webpage_bytes.decode('utf-8', 'replace')
 
-    def _webpage_read_content(self, urlh, url_or_request, video_id, note=None, errnote=None, fatal=True, prefix=None, encoding=None):
+    def _webpage_read_content(self, urlh, url_or_request, video_id, note=None, errnote=None, fatal=True,
+                              prefix=None, encoding=None, data=None):
         webpage_bytes = urlh.read()
         if prefix is not None:
             webpage_bytes = prefix + webpage_bytes
+        if isinstance(url_or_request, Request):
+            data = data if data is not None else url_or_request.data
```

Comment:
I think it might cleaner to do this, together with the below suggestion
(just fixing my own earlier suggestion)
```suggestion
        url_or_request = self._create_request(url_or_request, data)
```